fgspectra/models.py

```diff
@@ -1,89 +1,134 @@
-# -*- coding: utf-8 -*-
 r"""
-Foreground models.
+Models of cross-spectra
 
 This module draws inspiration from FGBuster (Davide Poletti and Josquin Errard)
 and BeFoRe (David Alonso and Ben Thorne).
 """
-import fgspectra.frequency, fgspectra.power
+from abc import ABC, abstractmethod
 import numpy as np
+from . import frequency as fgf
+from . import power as fgp
 
 
-class CompositeModel:
-    def __init__(self):
-        self.params = {}
-        return
+class CrossSpectrum(ABC):
+    """Base class for cross-spectra."""
 
-    def model(self, nu_i, nu_j, ell, **kwargs):
-        """Compute the model at frequency and ell combinations.
+    @abstractmethod
+    def __call__(self, *args):
+        """Make it callable."""
+        pass
+
+
+class FactorizedCrossSpectrum(CrossSpectrum):
```

Comment:
I changed from `CompositeModel` because I found "factorized" marginally more telling than "composite", happy to revert back to it (or to `CompositeCross`?) if people thing otherwise